azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -86,12 +107,86 @@ public ProjectManager(ProjectLoader loader, Props props) {
     new XmlValidatorManager(prop);
     loadAllProjects();
     loadProjectWhiteList();
+
+    CleanerThread cleanerThread = new CleanerThread();
```

Comment:
Instead of this why not use a scheduled threadpool executor with one thread?

Reply:
There were already some CleanerThread implementations in Azkaban, I followed the same pattern

Reply:
Ok